### PR TITLE
Add OpenAPI decoders endpoints

### PR DESCRIPTION
**Pull request by JmZero**
Hello team,

This PR adds the API endpoints /decoders in our OpenAPI yaml spec:

GET /decoders
GET /decoders/:decoder_name
GET /decoders/files
GET /decoders/parents

Best regards,
Jesús

### CODE REVIEW

File: api/api/controllers/decoders_controller.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    pass
+
+def get_decoders_by_name():
+    pass
+
+def get_decoders_files():
+    pass
+
+def get_decoders_parets():
+    pass
```
Comment (crd1985) [inline api/api/controllers/decoders_controller.py]: No empty end line.

File: api/api/controllers/decoders_controller.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+def get_decoders():
+    pass
+
+def get_decoders_by_name():
+    pass
+
+def get_decoders_files():
+    pass
+
+def get_decoders_parets():
```
Comment (crd1985) [inline api/api/controllers/decoders_controller.py]: Typo parets -> parents

File: api/api/spec/spec.yaml [hunk 3]
```diff
@@ -2726,8 +2844,254 @@ paths:
...
+                              type: array
+                              items:
+                                $ref: '#/components/schemas/Decoder'
+              example:
+                error: 0
+                data:
+                  totalItems: 571
+                  items:
+                  - file: 0005-wazuh_decoders.xml
+                    path: /var/ossec/ruleset/decoders
```
Comment (crd1985) [inline api/api/spec/spec.yaml]: Absolute paths cannot be returned to user. Please, modify this example to be realistic.

Comment (JmZero) [reply api/api/spec/spec.yaml]: Ok, I going to modify all return which include paths. Thanks.

**Pull request merged**